### ISSUE TITLE
ByteBuffer.Add(char) is Encoding dependant

### ISSUE THREAD

**Issue by ujos** [epam/fix-antenna-net-core#51]
IMO `ByteBuffer.Add(char)` method is implemented incorrectly. 

Following is the code:

```csharp
public ByteBuffer Add(char b)
{
    if (Buffer.Position == Buffer.Length)
    {
        IncreaseBuffer(64);
    }

    _writer.Write(b);
    return this;
}
```

Inside that function `BinaryWriter.Write(char)` method is used. According to [MSDN](https://docs.microsoft.com/en-us/dotnet/api/system.io.binarywriter.write?view=net-6.0#system-io-binarywriter-write(system-char))

> `BinaryWriter.Write(char)` writes a Unicode character to the current stream and advances the current position of the stream in accordance with the `Encoding` used and the specific characters being written to the stream.

Thus based on Encoding that method can write more than 1 byte (even 4 bytes) into buffer. Is it expected?

**Comment by OlehVasylievEPAM** [comment]
Unexpected issue. We will fix it in near future. Thanks!

**Comment by OlehVasylievEPAM** [comment]
Hello ujos, your issue is included in release 1.1.1. Please, let me know if you need a hotfix ASAP.

**Comment by OlehVasylievEPAM** [comment]
Hi ujos, the new release (1.2.0) is available. The release contains the fix. Thanks for your impact!
